### PR TITLE
Alchemiscale compute 2

### PR DESCRIPTION
**Pull request by richardjgowers**
@dotsdl I tried to write tests for your PR

### CODE REVIEW

File: gufe/protocols/protocoldag.py [hunk 1]
```diff
@@ -402,23 +401,16 @@ def execute_DAG(protocoldag: ProtocolDAG, *,
         # `ProtocolUnitResult`
         inputs = _pu_to_pur(unit.inputs, results)
 
-        if scratch_basedir is not None:
```
Comment (richardjgowers) [inline gufe/protocols/protocoldag.py:405]: @dotsdl the logic here seemed wrong.  If `scratch_basedir is not None`, then you probably don't want to wrap it as a `TemporaryDirectory` if you then want to keep it.

Also, `ProtocolUnit.execute()` **also** has logic for dealing with scratch=None, so we can omit the second branch.

**Comment by dwhswenson** [comment]
Note: As you're diving into some of this, let's keep in mind that #120 tells us we need to make changes that will change the API around `shared`. I'm about to put a PR up for that.

File: gufe/tests/test_execute_dag.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        my_id = inputs['identity']
+
+        with open(os.path.join(ctx.shared, f'unit_{my_id}_shared.txt'), 'w') as out:
+            out.write(f'unit {my_id} existed!\n')
+        with open(os.path.join(ctx.scratch, f'unit_{my_id}_scratch.txt'), 'w') as out:
+            out.write(f'unit {my_id} was here\n')
+
+        return {
+            'log': 'finished',
+        }
```
Comment (dwhswenson) [inline gufe/tests/test_execute_dag.py:22]: If the intent is to have a protocol unit that outputs a file to be saved, I think you'll need to include that as part of the return here.

**Comment by dotsdl** [comment]
Thanks @richardjgowers! Sorry for the delay; been focused downstream on `alchemiscale` bits that use this, but will loop back around to this once I get things tied off more there.

**Comment by dotsdl** [comment]
Merging this into #148 to hammer on in one place. Thanks @richardjgowers!

**Pull request merged**